include/abstract_filter_store.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+#include "common_includes.h"
+#include "utils.h"
+#include <any>
+
+namespace diskann
+{
+
+enum class FilterMatchStrategy
+{
+    SET_INTERSECTION
+};
+// This class is responsible for filter actions in index, and should not be used outside.
+template <typename label_type> class AbstractFilterStore
```

Comment:
Should we use the term "predicate" instead of filter? So we will have AbstractPredicateStore?